services/mbot_oled_display.py

```diff
@@ -4,22 +4,40 @@
 import time
 import qrcode
 import math
+import logging
 
 from luma.core.interface.serial import i2c
 from luma.core.render import canvas
 from luma.oled.device import ssd1306
 from PIL import ImageFont
 
-fontpath = str("/usr/local/etc/arial.ttf")
-font = ImageFont.truetype(fontpath, 14)
-fontpath = str("/usr/local/etc/arial.ttf")
-font_small = ImageFont.truetype(fontpath, 10)
-
-device = ssd1306(i2c(port=1, address=0x3C))
+# Setup logging
+log_file = "/var/log/mbot/mbot_oled.log"
+os.makedirs(os.path.dirname(log_file), exist_ok=True)
+logging.basicConfig(filename=log_file, level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
```

Comment:
I like the idea of using a log, but I recommend also publishing logging messages to the terminal so they are picked up by the service status messages and are visible on the terminal if the script is run. Here is an example of how this is done in MBot Bridge: https://github.com/mbot-project/mbot_bridge/blob/v1.0.0/src/mbot_bridge/server.py#L424-L432